File: iosApp/iosApp/ViewModel/SignUpViewModel.swift
//
//  SignUpViewModel.swift
//  iosApp
//
//  Created by Chinthaka on 05/04/2021.
//  Copyright © 2021 orgName. All rights reserved.
//

import Foundation
import shared


class SignUpViewModel {
    private let sharedViewModel: SharedViewModel = SharedViewModel()
    
    public func signUpUser(firstName : String,lastName : String,mobile : String , pass:String, completion: @escaping (ScreenState)->()) {
        let userRegistration = UserRegistration(username:mobile,password: pass,firstName: firstName,lastName: lastName)
        self.sharedViewModel.stateProvider.getUserRegistrationState(user: userRegistration) { (state : ScreenState) in
            completion(state)
        }
    }
}

File: androidApp/src/main/java/com/dev/kmmwallet/androidApp/viewmodel/SignUpViewModel.kt
package com.dev.kmmwallet.androidApp.viewmodel

import com.dev.kmmwallet.shared.datasource.webservice.model.request.UserRegistration
import com.dev.kmmwallet.shared.viewmodel.event.signup.getUserRegistrationState
import com.dev.kmmwallet.shared.viewmodel.launchCoroutine
import com.dev.kmmwallet.shared.viewmodel.status.ScreenState

class SignUpViewModel : BaseViewModel() {
    fun signUpUser(
        userName: String,
        pass: String,
        firstName: String,
        lastName: String,
        updateUi: (loginState: ScreenState) -> Unit
    ) {
        launchCoroutine {
            sharedViewModel.stateProvider.getUserRegistrationState(UserRegistration(userName, pass, firstName, lastName),
                fun(signUpState) {
                    updateUi(signUpState)
                })
        }
    }
}